### DISCUSSION TITLE
Ideas

### DISCUSSION THREAD

**Discussion by AbduazizZiyodov** [AbduazizZiyodov/docker-dashboard#1] [Ideas]
Write your ideas for this project, I really appreciate it :heart:

**Comment by loga4m** [comment]
Fikrim appni boshqa oilalarga tushadigan qilish haqida.
Configga dict qilib, keylarga oilalar nomini yozib, masalan arch based, ubuntu based keyin valuelarga har xil oilada xar xil bo'lgan command, pathlarni yozib chiqib, install.py da supported oilalarni raqamlab chiqarib, keyin userdan o'zining tizimini oilasi raqamini so'raydi va user kiritgandan so'ng family = user_input qilib oladi. Keyin komanda run qiladigan joylarda (configdan  supported_families(path,commandlarni store qiladigan dict)  import qilib olingan bo'ladi) 

`run_command(supported_families[family]["commands"][1])` shunaqa qiladi.

 Pathda 

`run_command(f"copy comammandlar /etc/{supported_families[family]['paths'][1]'})`

Menimcha fikrimni asosini tushunib oldingiz.
G'alati, juda sodda, oddiy yechim bo'lishi mumkin. Darajamga yarasha yechim.

**Comment by loga4m** [comment]
Assalamu alaykum! 
Can you add uninstall.py, if possible?
There are so many reasons. 
Although we can see what processes will go in the installation progress, App installation is abstract. 
After installation or copying files, if users get bugs or troubles, the one way will be reinstallation. Or, if the user wanted to change the version of the app, the user should uninstall the old version and install a new one. So, users may be simple users, who don't have deep knowledge about OS, which they're using. So, uninstalling manually will be very very hard. If you make an uninstall.py script for the solution to this problem, it will be good work.

For example, I've installed this app on the arch(I forgot, that this app is only for the ubuntu family). Yes, you are right, installation failed. Failed where? After all commands including copying files, when reached step "installing Debian packages". So, what I should do? Clean my system from this thing. It was so hard to clean because I don't have all knowledge used in this project.
